### PR TITLE
Adds Apple Fritters

### PR DESCRIPTION
**Pull request by TeomanTheGreat**
## About The Pull Request
Adds apple fritters. **Apple Fritter sprite by Kate Bishop** 

## Why It's Good For The Game
Apple fritters taste good and add more variety to what you see on the kitchen counter

## Changelog
:cl: Phil Smith
add: Apple fritters. 
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
bath salts but for botanists with minimal recoil

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -763,3 +763,21 @@
...
+	name = "Frittonium"
+	description = "The secret to the power of apple fritters"
+	reagent_state = LIQUID
+	color = "#e59b1f"
+	taste_description = "powerful"
+
+/datum/reagent/consumable/frittonium/on_mob_metabolize(mob/living/M)
+	to_chat(M, "<span class='notice'>You feel like a tank but the feeling is fading as every second goes by!</span>")
+	M.maxHealth += 100
+	M.health += 100
```
Comment (Kapu1178) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: No.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 1 above)
Comment (TeomanTheGreat) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: it lasts for like 4 seconds (or how much time it takes to metabolize the units in the fritters) and takes it away

**Comment by TeomanTheGreat** [comment]
> 
> 
> bath salts but for botanists with minimal recoil

its very easy to get fat eating these things so don't expect to be able to chow down a bunch of the stuff and have a bunch of frittonium in your body

**Comment by zeskorion** [comment]
Ugh this is dumb for a meme. This literally doubles your crit threshold

Meme code is bloody stupid

**Comment by BrumalLake** [comment]
i'm pretty sure it takes 20 seconds to metabolize 5 units even more when you're hungry as it's a consumable reagent and even longer with our average 60% td
just to get knocked down for 6 seconds regardless of how much you ate and there are a few species that can't get fat

**Comment by TeomanTheGreat** [comment]
> 
> 
> i'm pretty sure it takes 20 seconds to metabolize 5 units even more when you're hungry as it's a consumable reagent and even longer with our average 60% td just to get knocked down for 6 seconds regardless of how much you ate and there are a few species that can't get fat

ah that's right kinda forgot about those guys. oops. Have any ideas on how I can make this less busted while still being entertaining as a meme item?

**Comment by TeomanTheGreat** [comment]
> you're not gonna mediate this by saying that certain species not being able to get fat is the only problem with having double health

Double health for a very small time. I did test it and it didnt last all too long without it stopping. If this were to be going on during a fight then your opponent who ate this stuff wont be getting very far as soon as it wears off and all the damage they endured comes back to kick them in the ass. And plus you know the stam damage.

**Comment by BrumalLake** [comment]
aw man 6 seconds of down time when botanists can pack hundreds of units lasting minutes
you're right

**Comment by TeomanTheGreat** [comment]
> aw man 6 seconds of down time when botanists can pack hundreds of units lasting minutes
> you're right

Im sure you can cram quite a few of the plates in your bag or a tray but I'm not sure about eating it during battle especially when your being shot and or attacked back by whoever you were trying to rush after eating these. Also the ingredients require pastry base and the oil from the chefs oil tank so atleast they need to get the chef with them.

**Comment by BrumalLake** [comment]
wait this is about fritonium and you're talking about eating making you fat
i was taking about injecting pure fritonium this whole time
you won't even get fat like you said and you don't have to stand still to do it either
the recipe isn't that difficult
i don't believe that interaction with a sentient being in the room next to you gives a green light to have double health
i believe you can get the oil and eggs without them anyways

**Comment by TeomanTheGreat** [comment]
> i was taking about injecting pure fritonium this whole time



Oh. Now that i think about it i forgot to blacklist this on the random shit that botanists can add to plants. My mistake ill fix that right now.

**Comment by TeomanTheGreat** [comment]
> 
> 
> wait this is about fritonium and you're talking about eating making you fat i was taking about injecting pure fritonium this whole time you won't even get fat like you said and you don't have to stand still to do it either the recipe isn't that difficult i don't believe that interaction with a sentient being in the room next to you gives a green light to have double health i believe you can get the oil and eggs without them anyways

oh also I know this isn't my place to ask you this since this is my pr but do you have any suggestions of what I can add to the recipe to make this harder to obtain without it being stupidly difficult. Its like 11 pm where I am and I didn't get much sleep last night so my thinking isn't exactly on normal levels.

**Comment by BrumalLake** [comment]
if you put food in a multi mixer and press the button labeled grind a magical substance comes out nevertheless
and i don't believe in making it harder to get just cause it's broken
it should be the other way around

**Comment by TeomanTheGreat** [comment]
> if you put food in a multi mixer and press the button labeled grind a magical substance comes out nevertheless

Aww man you didn't read everything in the pr didn't you. Its set to ungrindable exactly to prevent people from cheesing it.

**Comment by BrumalLake** [comment]
yeah sorry for not scouring every line of code to see if you were trying to stealth remove pulse slugs again

**Comment by TeomanTheGreat** [comment]
> yeah sorry for not scouring every line of code to see if you were trying to stealth remove pulse slugs again

Do i have to explain it to you again. That was a fuckup in my commit history because i was a dumbass. Also this pr isnt even that long lol.

**Comment by BrumalLake** [comment]
that was a joke lads
i didn't even make a review you can't tell if i read it or not but i did skim
it should at least be in the description of a pr adding something far past the edge

**Comment by TeomanTheGreat** [comment]
> that was a joke lads

Hard to tell whats not a joke and what is when its on a github pr comment section

> it should at least be in the description of a pr adding something far past the edge

Done

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -763,3 +763,22 @@
 	if(prob(10))
 		M.say(pick("I hate my wife.", "I just want to grill for God's sake.", "I wish I could just go on my lawnmower and cut the grass.", "Yep, Quake. That was a good game...", "Yeah, my PDA has wi-fi. A wife I hate."), forced = /datum/reagent/consumable/char)
 	..()
+
+/datum/reagent/consumable/frittonium // literally stronger brave bull (with a side effect) and with zero tactical use unless you manage to eat it during a fight in which you will collapse seconds later
```
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Remove this

**Review by Archanial — CHANGES_REQUESTED** [review]

**Comment by ItsTisme** [comment]
Add a OD threshold, or do the thing most chems do where after X amount of ticks the bad sides creep up. Bassicly a hardcap on how long this effect lasts. Double health + any good armor is going to be pain to deal with

**Comment by Vexylius** [comment]
Adding content just for the meme is not a good idea

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (TeomanTheGreat) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: The comment or the chem itself

**Comment by TeomanTheGreat** [comment]
All right max health is lowered to 40 and there is an overdose effect now.

**Comment by Tsar-Salat** [comment]
> All right max health is lowered to 40 and there is an overdose effect now.

Imagine going to the bakery and you see a guy overdose on _**Apple Fucking Fritters**_

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (Archanial) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Chem itself

Comment (TeomanTheGreat) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: i mean considering i just nerfed it even further I would say its not necessary at the moment. Feel free to state your new opinion if you have one. If you still hate it ill get rid of it.

**Comment by Isy232** [comment]
> All right max health is lowered to 40 and there is an overdose effect now.

@TeomanTheGreat Adding features like +40 health for eating a food is an LRP mechanic. How the fuck are you supposed to say "Eating this gives me more health" IC, it just makes no sense.

I would say make Apple Fritters give a large amount of positive mood, which IIRC DOES impact crit threshold, so it would give you part of the meme mechanic without any problems.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -763,3 +763,26 @@
...
+
+/datum/reagent/consumable/frittonium/on_mob_metabolize(mob/living/M)
+	to_chat(M, "<span class='notice'>You feel like a tank but the feeling is fading as every second goes by!</span>")
+	M.maxHealth += 40
+	M.health += 40
+
+/datum/reagent/consumable/frittonium/on_mob_end_metabolize(mob/living/M)
+	to_chat(M, "<span class='notice'>You no longer feel like a tank and your muscles ache like all hell! What did you eat?!</span>")
+	M.maxHealth -= 40
+	M.health = min(M.health - 40, M.maxHealth) // this can indeed kill someone if they took a lot of damage during the fritter metabolization
```
Comment (Kapu1178) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: This is fucking stupid. It bypasses a ton of damage systems in the game, use take_overall_damage() or adjustToxLoss()

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 3 above)
Comment (TeomanTheGreat) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: No longer relevant

**Comment by TeomanTheGreat** [comment]
Fritters now add mood instead of health

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (TeomanTheGreat) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: no longer relevant. fritters now add mood.

**Comment by TeomanTheGreat** [comment]
> 
> 
> > All right max health is lowered to 40 and there is an overdose effect now.
> 
> @TeomanTheGreat Adding features like +40 health for eating a food is an LRP mechanic. How the fuck are you supposed to say "Eating this gives me more health" IC, it just makes no sense.
> 
> I would say make Apple Fritters give a large amount of positive mood, which IIRC DOES impact crit threshold, so it would give you part of the meme mechanic without any problems.

done

**Comment by BrumalLake** [comment]
please stop using the "makes no sense" argument 🤮 it doesn't even make no sense in this case either

**Comment by Crossedfall** [comment]
I like how your CL entry explicitly names you, but you didn't add in the credits for Kate. 😔

As for the change itself, it's a "funny" meme but the actual content is pretty weak. 🔑

**Comment by francinum** [comment]
This code's a mess, overrides species tastes without doing so properly, you can end up being disgusted but also 'loving it' at the same time.
🗝️

**Comment by Zanara04** [comment]
:)

**Pull request closed without merging**